assertk/src/jvmMain/kotlin/assertk/assertions/optional.kt

```diff
@@ -0,0 +1,66 @@
+package assertk.assertions
+
+import assertk.Assert
+import assertk.assertions.support.expected
+import assertk.assertions.support.show
+import com.willowtreeapps.opentest4k.AssertionFailedError
+import java.util.*
+
+/**
+ * Extracts an assertion against an guaranteed present value. If no value is present than
+ * an the assertion fail.
+ *
+ * Examples:
+ *
+ * ```
+ * val optional = Optional.of(12)
+ * assertThat(optional).value().isGreaterThan(11) // Passes
+ * assertThat(optional).value().isEqualTo(12)     // Passes
+ *
+ * val empty = Optional.empty<Int>()
+ * assertThat(empty).value().isInstanceOf(Int::class) // Fails
+ * assertThat(empty).value().isEqualTo(0) // Fails
+ *
+ * ```
+ */
+fun <T> Assert<Optional<T>>.value(): Assert<T> {
+    return transform { actual ->
+        actual.orElseThrow {
```

Comment:
As much as I like using `orElseThrow` here, I think we should keep the messages consistent and use an `expected()` call instead.

Reply:
Will change this. Thanx for the feedback. 👍🏼